docs/Building SpiderMonkey.md

```diff
@@ -0,0 +1,62 @@
+# Building SpiderMonkey for Embedders #
+
+Use these instructions to build your own copy of SpiderMonkey.
+
+## Prerequisites ##
+
+You will need a **C++ compiler** that can handle the C++14 standard,
+**GNU Make**, **zlib**, and **libffi**.
+These can usually be installed with a package manager.
+You will also need **ICU** of at least version 59.1.
+ICU can also usually be installed with a package manager, but the
+version might not be new enough.
+
+> **NOTE** If you can't install a new enough version of ICU, you can
+> build the copy bundled with SpiderMonkey by omitting
+> `--with-system-icu` in the build instructions below, at the cost of a
+> longer build time.
+
+## Getting the source code ##
+
+Currently, the most reliable way to get the SpiderMonkey source code is
+to download the Firefox source.
+At the time of writing, the latest source for Firefox ESR 60, which
+contains the source for SpiderMonkey ESR 60, can be found here:
+http://ftp.mozilla.org/pub/firefox/releases/60.3.0esr/source/
+
```

Comment:
A blurb about ESR:

The ESR releases have a major release approximately once a year with security patches released throughout the year. It is recommended that embedders track ESR to have reasonable API stability. The master branch of SpiderMonkey experiences a fair amount of breaking changes unfortunately, driven by the needs of the Firefox browser.